### DISCUSSION TITLE
How about if work on React v18 concurrently mode?

### DISCUSSION THREAD

**Discussion by rwu823** [pmndrs/jotai#1078] [Q&A] [answered]
Could jotai handle it well?

**Comment by dai-shi** [comment, marked as answer]
Short Answer: Yes, absolutely.

Long Answer: There are a few technical limitations, but in general they are very edge cases. It's unlikely you hit them, and if you do and have any issues, please report them. I've been experimenting CM a lot. See this if interested: https://github.com/dai-shi/will-this-react-global-state-work-in-concurrent-rendering. On the other hand, server side rendering with Suspense is new to me. There are no known issues at the moment, but it's still open space.

**Comment by dai-shi** [comment]
For someone interested: https://twitter.com/dai_shi/status/1509533380786155525
